### PR TITLE
Feature/3108 refactor CustomEvent handlers in app share folder

### PR DESCRIPTION
**Pull request by metzgegu**
#3108 

## Checkpoints

The goal of this section is to help code reviewers and the merge request author to do the required checks on this pull request. Please don't edit this list.
If one or more of these checkpoints are out of scope, please write below why they are.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, then there are associated issues and issue is referenced in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present
- [x] Automated tests have been written (covering the feature or non regression), or *at least* an issue has been created to implement the test

**For developers**

- [x] Manual tests done to ensure stability on whole application layers, issue expectation follow
- [x] The original issue has been updated with a short summary of the implemented solution

**For the quality team**

- [x] Tested by the quality team

### CODE REVIEW

File: frontend_app_share_folder_advanced/src/container/ShareFolderAdvanced.jsx [hunk 1]
```diff
@@ -55,29 +56,36 @@ class ShareFolderAdvanced extends React.Component {
...
+      { name: CUSTOM_EVENT.HIDE_APP(this.state.config.slug), handler: this.handleHideApp },
+      { name: CUSTOM_EVENT.ALL_APP_CHANGE_LANGUAGE, handler: this.handleAllAppChangeLanguage }
+    ])
   }
 
-  customEventReducer = ({ detail: { type, data } }) => {
+  // Custom Event Handlers
+  handleShowApp = data => {
     const { props, state } = this
+    console.log('%c<ShareFolderAdvanced> Custom event', 'color: #28a745', CUSTOM_EVENT.SHOW_APP, data)
```
Comment (gbonaspetti) [inline frontend_app_share_folder_advanced/src/container/ShareFolderAdvanced.jsx:69]:
```suggestion
    console.log('%c<ShareFolderAdvanced> Custom event', 'color: #28a745', CUSTOM_EVENT.SHOW_APP(state.config.slug), data)
```

File: frontend_app_share_folder_advanced/src/container/ShareFolderAdvanced.jsx [hunk 2]
```diff
@@ -55,29 +56,36 @@ class ShareFolderAdvanced extends React.Component {
...
-        this.setHeadTitle()
-        break
-    }
+    props.appContentCustomEventHandlerShowApp(data.content, state.content, this.setState.bind(this), () => {})
+    this.setHeadTitle()
+  }
+
+  handleHideApp = data => {
+    const { props } = this
+    console.log('%c<ShareFolderAdvanced> Custom event', 'color: #28a745', CUSTOM_EVENT.HIDE_APP, data)
```
Comment (gbonaspetti) [inline frontend_app_share_folder_advanced/src/container/ShareFolderAdvanced.jsx:77]:
```suggestion
    console.log('%c<ShareFolderAdvanced> Custom event', 'color: #28a745', CUSTOM_EVENT.HIDE_APP(this.state.config.slug), data)
```

File: frontend_app_share_folder_advanced/test/.mocharc.js [hunk 3]
```diff
@@ -6,7 +6,9 @@ module.exports = {
     'regenerator-runtime/runtime',
     '@babel/register',
     'ignore-styles',
-    'test/setup'
+    'test/setup',
+    // INFO - CH - 2020-06-05 - Isomorphic-fetch provide the fetch API because Node does not contain it, and unit test run in Node context
```
Comment (gbonaspetti) [inline frontend_app_share_folder_advanced/test/.mocharc.js:10]: CH?

Comment (Skylsmoi) [reply frontend_app_share_folder_advanced/test/.mocharc.js]: haha, I guess he just copast the comment I made

**Pull request merged**